src/components/views/room_settings/PublishedAliases.js

```diff
@@ -0,0 +1,332 @@
+     * local aliases that are not already in the alt alias list. */
+    const altSuggestions = localAliases && localAliases.filter((alias) =>
+      !state.alt.working.includes(alias),
+    ).map((alias) => alias.replace(/^#/, '')); // strip the # for display
+
+    /* Canonical aliases can be either existing local aliases or existing alt aliases */
+    const canonicalSuggestions = Array.from(new Set([...state.alt.working, ...(localAliases || [])]));
+    const explicitlyInclude = (state.canonical.working && !canonicalSuggestions.includes(state.canonical.working));
+
+    return (
+        <OverlaySpinner active={!!state.submitting}>
+            <span className='mx_SettingsTab_subheading'>{_t("Published Addresses")}</span>
+            <p>{_t("Published addresses can be used by anyone on any server to join your room. " +
+                "To publish an address, it needs to be set as a local address first.")}</p>
+            <Field onChange={(ev) => onSetCanonical(ev.target.value)} value={state.canonical.working || ''}
+                   disabled={!canSetCanonicalAlias || !!state.submitting}
+                   element='select' id='canonicalAlias' label={_t('Main address')}>
+                <option value="" key="unset">{ _t('not specified') }</option>
+                {canonicalSuggestions.map((alias, i) =>
+                  <option value={alias} key={i}>{alias}</option>)}
+                {explicitlyInclude && <option key="explicit" value={state.canonical.working}>
+                    {state.canonical.working}</option>}
+            </Field>
+            <RoomPublishSetting roomId={room.roomId} canSetCanonicalAlias={canSetCanonicalAlias} />
+            <datalist id="mx_AliasSettings_altRecommendations">
+                {altSuggestions && altSuggestions.map(alias =>
+                    <option value={alias} key={alias} />,
+                )}
+            </datalist>
+            <EditableAltAliasList
+                canEdit={canSetCanonicalAlias}
+                canRemove={canSetCanonicalAlias || !!state.submitting}
+                className="mx_RoomSettings_altAliases"
+                error={error}
+                id="roomAltAliases"
+                items={state.alt.working}
+                itemsLabel={_t('Other published addresses:')}
+                newItem={state.newAltAlias}
+                noItemsLabel={canSetCanonicalAlias ?
+                    _t('No other published addresses yet, add one below') :
+                    _t('No other published addresses yet.')}
+                onItemAdded={onAddAlias}
+                onItemRemoved={onRemoveAlias}
+                onNewItemChanged={(value) => dispatch({value, type: UPDATE_ALT_ALIAS})}
+                placeholder={_t('New published address (e.g. #alias:server)')}
+                suggestionsListId="mx_AliasSettings_altRecommendations"
+            />
+        </OverlaySpinner>);
+}
+
+export default undefined;
```

Comment:
what is the need for this?

Reply:
the compiler was complaining, i'll see if removing it helps...